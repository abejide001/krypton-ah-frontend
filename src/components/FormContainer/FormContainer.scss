@import "../../styles/partials/styles";

.ui {
  &.modal {
    border: 2px solid $black;
    border-radius: 0.938rem;

    &.authModal {
      margin: 1rem auto;
      padding: 1.25rem;
    }

    form {
      border-radius: 0;
    }
  }
}

.ui.basic.button {
  border-radius: 0;

  margin: {
    bottom: 3.125rem;
    top: 1.875rem;
  }

  &:hover {
    background: $primarycolor !important;
    color: $white !important;
    transition: 0.5s ease-in-out;
  }
}

.headerTitle {
    border-bottom: 0;
    color: lighten($primarycolor, 53%);

    font: {
      size: 1.5rem;
      weight: 300;
    }

    margin-top: 1.25rem;
    text-transform: uppercase;
    
  }

.headerBotton {
  border: 0.125rem solid $black;

  margin: {
    bottom: 3.875rem;
    left: 0;
    top: -0.437rem;
  }

  width: 3.3rem;
}

h5.baseText {
  font: {
    size: 0.813rem;
    weight: 200;
  }

  margin: {
    bottom: 1.875rem;
    top: 3.125rem;
  }

  text-align: center;

  a {
    color: $red;
  }

  :hover {
    color: $primarycolor
  }
}
@media (min-width: 768px) and (max-width: 767px) {
  .ui {
    &.modal {

      &.authModal {
        width:80%;
      }
    }
  }   
}
@media (max-width: 340px) {
  .ui {
    &.modal {
      
      &.authModal {
        width:100%;
      }
    }
  }   
}