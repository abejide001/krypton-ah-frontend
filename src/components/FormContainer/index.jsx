import React, { Fragment } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { Modal } from 'semantic-ui-react';
import SocialIcons from '../SocialComponent';
import ModalDivider from '../ModalDivider';
import './FormContainer.scss';

class ModalComponent extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      open: true
    };
    this.handleClose = this.handleClose.bind(this);
  }

  handleClose() {
    this.setState({ open: false });
  }

  render() {
    const {
      title, baseText, link, formLink, children
    } = this.props;
    const { open } = this.state;
    return (
      <Modal size="mini" open={open} onClose={this.handleClose} className="authModal">
        <Modal.Content>
          <h2 className="header headerTitle">{title}</h2>
          <hr className="headerBotton" />
          <Modal.Description>
            <Fragment>{children}</Fragment>
            <ModalDivider text={`or ${title} with`} />
            <SocialIcons />
            <h5 className="baseText">
              {baseText}
              &nbsp;
              <Link to={`/${formLink}`}>
                {link}
              </Link>
            </h5>
          </Modal.Description>
        </Modal.Content>
      </Modal>
    );
  }
}

ModalComponent.propTypes = {
  title: PropTypes.string,
  baseText: PropTypes.string,
  link: PropTypes.string,
  formLink: PropTypes.string,
  children: PropTypes.node
};

ModalComponent.defaultProps = {
  title: 'sign in',
  baseText: 'Create an account',
  link: 'sign up',
  formLink: 'signin',
  children: null
};

export default ModalComponent;
