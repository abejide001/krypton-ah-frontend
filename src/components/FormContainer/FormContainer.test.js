import React from 'react';
import sinon from 'sinon';
import { shallow } from 'enzyme';
import FormContainer from '.';

describe('Form Container component', () => {
  it('should matches the snapshot', () => {
    const wrapper = shallow(<FormContainer />);
    expect(wrapper).toMatchSnapshot();
  });
  it('should simulate on click', () => {
    const wrapper = shallow(<FormContainer />);
    wrapper.state = {
      open: true
    };
    const handleCloseSpy = sinon.spy(wrapper.instance(), 'handleClose');
    wrapper.update();
    wrapper.find('Modal').simulate('click');
    wrapper.setState({ open: true });
    expect(handleCloseSpy.calledOnce).toBeDefined();
  });
});
