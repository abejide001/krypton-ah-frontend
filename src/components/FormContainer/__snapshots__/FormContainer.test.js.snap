// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Container component should matches the snapshot 1`] = `
<Fragment>
  <RedirectLoader />
  <Modal
    centered={true}
    className="authModal"
    closeOnDimmerClick={true}
    closeOnDocumentClick={false}
    dimmer={true}
    eventPool="Modal"
    onClose={[Function]}
    open={true}
    size="mini"
  >
    <ModalContent>
      <h2
        className="header headerTitle"
      >
        sign in
      </h2>
      <hr
        className="headerBotton"
      />
      <ModalDescription>
        <ModalDivider
          text="or sign in with"
        />
        <SocialIcons />
        <h5
          className="baseText"
        >
          Create an account
          Â 
          <Link
            replace={false}
            to="/signin"
          >
            sign up
          </Link>
        </h5>
      </ModalDescription>
    </ModalContent>
  </Modal>
</Fragment>
`;
